src/test/java/uk/gov/gchq/palisade/UtilTest.java

```diff
@@ -55,12 +58,15 @@ public void shouldUpdateRecord() {
     public void shouldUpdateRecordFromAllRules() {
         //given
         final Rules<String> rules = new Rules<String>()
-                .addRule("r1", (record, user, context) -> "fromRule")
+                .addRule("r1", (record, user, context) -> record.concat("fromRule"))
                 .addRule("r2", (record, user, context) -> record.concat("2ndRule"));
         //when
         final String actual1 = applyRulesToItem("String", null, null, rules);
         //then
-        assertEquals("'fromRule2ndRule' should be returned as the record has been updated for all rules", "fromRule" + "2ndRule", actual1);
+        assertThat("item should have all rules applied", actual1, allOf(
+                containsString("fromRule"),
+                containsString("2ndRule"),
+                containsString("String")));
     }
```

Comment:
This change was because there is no guarantee of ordering of rule applications, so record.concat(a).concat(b) could equally be record.concat(b).concat(a).
1. Should this pattern be applied to other tests, which only pass because of "lucky" hashcode ordering?
2. Will this be a cause for future concern? Do we need to document that rule order is not enforceable? (This is probably obvious in practice, but could lead to "bugs" in applied policy and expected output)

Reply:
I thought we had the rules in a sorted collectable type, so it should always be in a certain order

Reply:
revert changes, rules should be applied in the order they were added to the LinkedHashMap